### ISSUE TITLE
[Discuss] Index patterns in discover

### ISSUE THREAD

**Issue by stacey-gammon** [stacey-gammon/kibana-dependency-tracking#62]
TLDR; currently, when viewing data in Discover app, user selects a single Kibana index pattern (defined in management app). Maybe Discover could extend this to allow selecting multiple index patterns and/or any Elasticsearch multi index API strings.

---

![image](https://user-images.githubusercontent.com/9773803/85843055-e9992680-b7a0-11ea-804c-d34c1cdaacae.png)

In the above screenshot you can see a single Kibana index pattern selected. This thread is to discuss possible enhancements to that. For example, instead of single Kibana index pattern user could be presented with a combobox where they can select multiple Kibana index patterns at once. Also, maybe user could be presented with an input line where they could enter arbitrary Elasticsearch multi index API string.

## Problem

- Not possible to view data from more than one Kibana index pattern at once.
- User must create an index pattern if they have none. Also, they must create an index pattern if none exists for the index they want to view.
- Dashboard visualizations can be powered by multiple Kibana index patterns (for example in case of Lens). Other Dashboard visualizations are not even powered by Kibana index patterns, but they use directly Elasticsearch multi index API, such as TSVB and Timelion. Hence, for Dashboard visualizations that are powered by something more than a single Kibana index pattern we cannot navigate to Discover using "Explore underlying data" action or a drilldown to Discover.

## Possible solutions

- Maybe Discover could support multiple selected Kibana index patterns at once.
- Maybe a *virtual* Kibana index pattern could be created when user navigates to Discover. That virtual Kibana index pattern would be represented by the same structure as actual index patterns, say `IIndexPattern` interface, but it would not be stored, it would just be created on the fly.
- Maybe Discover could support, in addition to Kibana index patterns, an arbitrary string input where user could enter Elasticsearch multi index API string.

## Challenges

I'm sure there are plenty of challenges to extend Discover app in such a way and I hope you will post some below, here is one:

- Multiple index patterns could have same fields but different field formats.

/cc @elastic/kibana-app @elastic/kibana-app-arch
